docs/tutorials/tutorial-unnest-datasource.md

```diff
@@ -0,0 +1,670 @@
+      },
+      "inputFormat": {
+        "type": "json"
+      }
+    },
+    "tuningConfig": {
+      "type": "index_parallel",
+      "partitionsSpec": {
+        "type": "dynamic"
+      }
+    },
+    "dataSchema": {
+      "dataSource": "nested_data2",
+      "granularitySpec": {
+        "type": "uniform",
+        "queryGranularity": "NONE",
+        "rollup": false,
+        "segmentGranularity": "YEAR"
+      },
+      "timestampSpec": {
+        "column": "t",
+        "format": "auto"
+      },
+      "dimensionsSpec": {
+        "dimensions": [
+          "dim1",
+          "dim2",
+          "dim3"
+        ]
+      },
+      "metricsSpec": [
+        {
+          "name": "m1",
+          "type": "floatSum",
+          "fieldName": "m1"
+        },
+        {
+          "name": "m2",
+          "type": "doubleSum",
+          "fieldName": "m2"
+        }
+      ]
+    }
+  }
+}
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+
+### Unnest `nested_data2`
+
+The following query performs two unnests. It unnests `dim3` into a column named `unnest-dim3`. It also performs an unnest on `dim2` and outputs the results to `unnest-dim2`. You can then treat the combination of `unnest-dim3` and `unnest-dim2` as Cartesian products.
+
+When you run the query, pay special attention to how the total number of rows has grown drastically. The source data has 2 rows. The unnested data has 12 rows, (2 x 2) + (2 x 4).
+
+<details><summary>Show the query</summary>
+
+```json
+{
+  "queryType": "scan",
+  "dataSource": {
+    "type": "unnest",
+    "base": {
+      "type": "unnest",
+      "base": {
+        "type": "table",
+        "name": "nested_data2"
+      },
+      "column": "dim3",
+      "outputName": "unnest-dim3",
+      "allowList": []
+    },
+    "column": "dim2",
+    "outputName": "unnest-dim2",
+    "allowList": []
+  },
+  "intervals": {
+    "type": "intervals",
+    "intervals": [
+      "-146136543-09-08T08:23:32.096Z/146140482-04-24T15:36:27.903Z"
+    ]
+  },
+  "limit": 1000,
+  "columns": [
+    "__time",
+    "dim1",
+    "dim2",
+    "dim3",
+    "m1",
+    "m2",
+    "unnest-dim3",
+    "unnest-dim2"
+  ],
+  "legacy": false,
+  "granularity": {
+    "type": "all"
+  },
+  "context": {
+    "debug": true,
+    "useCache": false
+  }
+}
+```
+
+</details>
+
+## Unnest an inline datasource
+
+You can also use the `unnest` datasource to unnest an inline datasource. The following query takes the row `[1,2,3]` in the column `inline_datasource` that is provided inline within the query and returns it as unnested values in the `output` column:
+
+<details><summary>Show the query</summary>
+
+```json
+{
+  "queryType": "scan",
+  "dataSource": {
+    "type": "unnest",
+    "base": {
+      "type": "inline",
+      "columnNames": [
+        "inline_datasource"
```

Comment:
Minor we should not call this column as `inline_datasource` as columns and datasource are different IMO